docs/src/main/sphinx/develop/connectors.rst

```diff
@@ -82,30 +82,158 @@ a :ref:`connector-page-sink-provider`.
 .. _connector-split-manager:
 
 ConnectorSplitManager
-^^^^^^^^^^^^^^^^^^^^^
+---------------------
 
 The split manager partitions the data for a table into the individual
 chunks that Trino will distribute to workers for processing.
 For example, the Hive connector lists the files for each Hive
-partition and creates one or more split per file.
+partition and creates one or more splits per file.
 For data sources that don't have partitioned data, a good strategy
 here is to simply return a single split for the entire table. This
 is the strategy employed by the Example HTTP connector.
 
 .. _connector-record-set-provider:
 
 ConnectorRecordSetProvider
-^^^^^^^^^^^^^^^^^^^^^^^^^^
+--------------------------
 
 Given a split and a list of columns, the record set provider is
 responsible for delivering data to the Trino execution engine.
+
 It creates a ``RecordSet``, which in turn creates a ``RecordCursor``
 that is used by Trino to read the column values for each row.
 
+The provided record set must only include requested columns
+in the correct order. If it returns more rows than necessary,
```

Comment:
"the correct order" is too vague. Make it more explicit by explaining how it relates to the order of the columns in the `getRecordSet()` call.

Reply:
Thanks! I added a sentence about the list of column handles passed to `getRecordSet()`.